examples/release_highlights/plot_release_highlights_0_24_0.py

```diff
@@ -0,0 +1,231 @@
+# <https://doi.org/10.3115/981658.981684>`_ can now be used with any
+# classifier that implements :term:`predict_proba`. The sub-classifier
+# will behave as a
+# semi-supervised classifier, allowing it to learn from unlabeled data.
+# Read more in the :ref:`User guide <self_training>`.
+
+import numpy as np
+from sklearn import datasets
+from sklearn.semi_supervised import SelfTrainingClassifier
+from sklearn.svm import SVC
+rng = np.random.RandomState(42)
+iris = datasets.load_iris()
+random_unlabeled_points = rng.rand(iris.target.shape[0]) < 0.3
+iris.target[random_unlabeled_points] = -1
+svc = SVC(probability=True, gamma="auto")
+self_training_model = SelfTrainingClassifier(svc)
+self_training_model.fit(iris.data, iris.target)
+
+##############################################################################
+# New SequentialFeatureSelector transformer
+# -----------------------------------------
+# A new iterative transformer to select features is available:
+# :class:`~sklearn.feature_selection.SequentialFeatureSelector`.
+# Sequential Feature Selection can add features once at a time (forward
+# selection) or remove features from the list of the available features
+# (backward selection), based on a cross-validated score maximization.
+# See the :ref:`User Guide <sequential_feature_selection>`.
+
+from sklearn.feature_selection import SequentialFeatureSelector
+from sklearn.neighbors import KNeighborsClassifier
+from sklearn.datasets import load_iris
+
+X, y = load_iris(return_X_y=True, as_frame=True)
+feature_names = X.columns
+knn = KNeighborsClassifier(n_neighbors=3)
+sfs = SequentialFeatureSelector(knn, n_features_to_select=2)
+sfs.fit(X, y)
+print("Features selected by forward sequential selection: "
+      f"{feature_names[sfs.get_support()]}")
+
+##############################################################################
+# New PolynomialCountSketch kernel approximation function
+# -------------------------------------------------------
+# The new :class:`~sklearn.kernel_approximation.PolynomialCountSketch`
+# approximates a polynomial expansion of a feature space when used with linear
+# models, but uses much less memory than
+# :class:`~sklearn.preprocessing.PolynomialFeatures`, and is often faster than
+# using a polynomial kernel in SVM.
+
+from sklearn.datasets import fetch_covtype
+from sklearn.pipeline import make_pipeline
+from sklearn.model_selection import train_test_split
+from sklearn.kernel_approximation import PolynomialCountSketch
+from sklearn.svm import LinearSVC
+
+X, y = fetch_covtype(return_X_y=True)
+pipe = make_pipeline(PolynomialCountSketch(degree=2, n_components=1000),
+                     LinearSVC())
+X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=5000,
+                                                    test_size=10000,
+                                                    random_state=42)
+pipe.fit(X_train, y_train)
```

Comment:
I did some experiments to try to avoid the convergence problem on this example while allowing the model to converge quickly to a model better than the baseline linear model.

- first it's important to scale the data with MinMaxScaler (it works much better than StandardScaler, I am not sure exactly why it's the case but that's the case).
- also let's use LogisticRegression instead of LinearSVC because it's more efficient to use the multinomial loss instead of the One vs Rest strategy of LinearSVC on multiclass classification problems (there are 6 classes in this dataset)
- finally and perhaps counter intuitively, we should limit n_components to 300 or similar. Beyond that we get the convergence problem again: I suspect the optimization problem becomes too ill-conditioned but this is just an intuition.

When doing all of the above with `max_iter=1000` for LogisticRegression, I get a test accuracy of 0.738 in 13s on my laptop.


Also I think it would be interesting to report the test accuracy in the code snippet:

```suggestion
pipe.fit(X_train, y_train).score(X_test, y_test)
```

and contrast that with a linear pipeline as the baseline:

```python
linear_baseline = make_pipeline(MinMaxScaler(), LogisticRegression)
linear_baseline.fit(X_train, y_train).score(X_test, y_test)
```

Reply:
Indeed conditioning is probably the cause.

If I add `PCA(n_components=100, whiten=True)` after `PolynomialCountSketch(degree=2, n_components=300)` it's converging even faster (and the test accuracy stays approximately the same).

We probably don't need to mention conditioning issues and PCA in the release highlights but it's interesting to know (and we could potentially improve the example and the doc later).

Reply:
Thanks @ogrisel experiencing outside basic parameters was beyond my understanding of the example.

Reply:
Note for later: the conditioning issue caused by `PolynomialCountSketch` based preprocessing might be interesting to consider when we finally have the time to work on https://github.com/scikit-learn/scikit-learn/pull/15583.

Reply:
> Thanks @ogrisel experimenting outside basic parameters was beyond my understanding of the example.

The fact that this preprocessor is tricky to make it work reveals usability problems.  Having linear models with default solvers that are more robust to conditioning issues as linked above is probably a promising avenue for improvement in scikit-learn.

Maybe we could also investigate if we could automatically cases where StandardScaler is likely to yield bad scaling and issue a warning in those cases. However I am not sure if it's easy to do without introducing annoying false positives.